feature-swap-impl/src/main/java/io/novafoundation/nova/feature_swap_impl/domain/validation/SufficientBalanceConsideringNonSufficientAssetsValidation.kt

```diff
@@ -0,0 +1,41 @@
+package io.novafoundation.nova.feature_swap_impl.domain.validation
+
+import io.novafoundation.nova.common.validation.ValidationStatus
+import io.novafoundation.nova.common.validation.valid
+import io.novafoundation.nova.common.validation.validOrError
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.AssetSourceRegistry
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.existentialDepositInPlanks
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.isSelfSufficientAsset
+import io.novafoundation.nova.runtime.ext.isCommissionAsset
+
+class SufficientBalanceConsideringNonSufficientAssetsValidation(
+    private val assetSourceRegistry: AssetSourceRegistry
+) : SwapValidation {
+
+    override suspend fun validate(value: SwapValidationPayload): ValidationStatus<SwapValidationFailure> {
+        val assetIn = value.detailedAssetIn.asset
+        val assetOut = value.detailedAssetOut.asset
+        val amount = value.detailedAssetIn.amountInPlanks
+
+        val isSelfSufficientAssetOut = assetSourceRegistry.isSelfSufficientAsset(assetOut.token.configuration)
+
+        if (!isSelfSufficientAssetOut && assetIn.token.configuration.isCommissionAsset) {
+            val existentialDeposit = assetSourceRegistry.existentialDepositInPlanks(value.detailedAssetIn.chain, assetOut.token.configuration)
+            val fee = value.swapFee.networkFee.amount
```

Comment:
Should we consider total fee, not only the network one? In particular, toBuyMinimumBalance part seems be relevant here as well

Reply:
If I understand everything correctly, then we do not need to use totalDeductedPlanks, and also take into account toBuyMinimumED, since this is a narrow case only for cases when assetIn == native asset, and we will pay commission only in it

Reply:
Another case we probably should consider is cross swapping when we swap `non native asset from chain1 to non sufficient asset from chain2` and `native asset balance` in `chain2` is `0`

Reply:
Yeah agree that toBuyMinimumED is not relevant here